core/src/test/java/cocktailofferer/service/image/query/ImageMetadataQueryServiceTest.java

```diff
@@ -0,0 +1,46 @@
+package cocktailofferer.service.image.query;
+
+import cocktailofferer.service.image.ImageMetadata;
+import cocktailofferer.service.image.ImageMetadataRepository;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.ContextConfiguration;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@DataJpaTest
+@ContextConfiguration(classes = ImageMetadataQueryServiceTestConfiguration.class)
+public class ImageMetadataQueryServiceTest {
+    @Autowired
+    private ImageMetadataQueryService imageMetadataQueryService;
+    @Autowired
+    private ImageMetadataRepository imageMetadataRepository;
+
+    @Test
+    public void givenImageMetadata_whenGetById_thenCorrectMetadataReturned() {
+        var imageMetadata = createImageMetadata();
+
+        var actualImageMetadata = imageMetadataQueryService.getById(imageMetadata.getId());
+
+        assertThat(actualImageMetadata.id()).isEqualTo(imageMetadata.getId());
+        assertThat(actualImageMetadata.name()).isEqualTo(imageMetadata.getName());
+        assertThat(actualImageMetadata.key()).isEqualTo(imageMetadata.getImageKey());
+    }
+
+    private ImageMetadata createImageMetadata() {
+        var imageMetadata = ImageMetadata.builder()
+                .id(1)
+                .imageKey("key")
+                .name("name")
+                .build();
+        return imageMetadataRepository.save(imageMetadata);
+    }
+
+    @Test
+    public void givenImageMetadata_whenGetById_thenCorrectExceptionThrown() {
```

Comment:
What do you mean `givenImageMetadata`?

Reply:
ImageMetadata created in database

Reply:
But it's not "created in database", nothing is created, the test is about getting nonexistent entity as I understand

Reply:
Yes, u are right
I will rename to givenNonexistentImageMetadataId